### PR TITLE
fix: add applicationCodes

### PR DESCRIPTION
**Pull request by 4886Talga**
Please review changes and come with your opinion what should be changed

### CODE REVIEW

File: server/apiCalls/kursPmDataApi.js [hunk 1]
```diff
@@ -15,9 +15,10 @@ async function getSortedAndPrioritizedMiniMemosWebOrPdf(courseCode) {
   try {
     const uri = client.resolve(paths.getPrioritizedWebOrPdfMemosByCourseCode.uri, { courseCode })
 
-    const { body } = await client.getAsync({ uri })
-    if (!body) log.debug('kurs-pm-data-api is not available at the moment in getSortedAndPrioritizedMiniMemosWebOrPdf')
-    return body || []
+    const res = await client.getAsync({ uri })
```
Comment (lenaar) [inline server/apiCalls/kursPmDataApi.js:18]: Previous code is more readable and follows best practices. Therefore it has to be changed as it was before. Destructure direct in await, `const { body } = await client.getAsync({ uri })`

File: test/mocks/mockCourseAnalysis.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-const mockCourseAnalysis = (semester = '20172', roundIdList = '1') => ({
+const mockCourseAnalysis = (semester = '20172', roundIdList = '1', applicationCodes = '1') => ({
```
Comment (lenaar) [inline test/mocks/mockCourseAnalysis.js:1]: `roundIdList` has to be removed, in order to be sure tests work

File: test/mocks/mockCourseAnalysis.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ const mockCourseAnalysis = (semester = '20172', roundIdList = '1') => ({
   pmFileName: 'pm-EI1220HT2017_1.pdf',
   analysisFileName: 'analysis-EI1220HT2017_1.pdf', // prev file from api
   roundIdList,
```
Comment (lenaar) [inline test/mocks/mockCourseAnalysis.js:11]: `roundIdList` has to be removed, in order to be sure tests work

File: test/mocks/mockMiniMemos.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ const mockedMiniMemosPdfAndWeb = {
       3: {
         _id: 'EI1220_20182_3',
         courseCode: 'EI1220',
+        applicationCodes: ['3'],
         changedBy: 'krisa',
         // koppsRoundId: '3',
         ladokRoundIds: ['3'],
```
Comment (lenaar) [inline test/mocks/mockMiniMemos.js:10]: `ladokRoundIds` has to be removed, in order to be sure tests work

File: test/mocks/mockMiniMemos.js [hunk 5]
```diff
@@ -14,18 +15,19 @@ const mockedMiniMemosPdfAndWeb = {
...
+      }
     },
     20172: {
       1: {
         _id: 'EI1220_20172_1',
         courseCode: 'EI1220',
+        applicationCodes: ['1'],
         changedBy: 'zzzz',
         // koppsRoundId: '1',
         ladokRoundIds: ['1'],
```
Comment (lenaar) [inline test/mocks/mockMiniMemos.js:33]: `ladokRoundIds ` has to be removed, in order to be sure tests work

File: test/mocks/mockMiniMemos.js [hunk 6]
```diff
@@ -37,13 +39,14 @@ const mockedMiniMemosPdfAndWeb = {
...
+        lastChangeDate: '2021-07-07T14:35:31.507Z'
+      }
     },
     20192: {
       1: {
         _id: 'EI1220_20192_1',
         courseCode: 'EI1220',
+        applicationCodes: ['1'],
         changedBy: 'zzzz',
         ladokRoundIds: ['1'],
```
Comment (lenaar) [inline test/mocks/mockMiniMemos.js:51]: `ladokRoundIds ` and `roundIdList` has to be removed everywhere, in order to be sure tests work

File: test/mocks/rawAnalysisData.js [hunk 7]
```diff
@@ -48,6 +49,7 @@ const mockRawAnalysisData = [
     pmFileName: '',
     analysisFileName: 'analysis-SF1624HT2018_9.pdf',
     roundIdList: '9',
```
Comment (lenaar) [inline test/mocks/rawAnalysisData.js:51]: `ladokRoundIds ` and `roundIdList` has to be removed everywhere, in order to be sure tests work

File: test/mocks/transformedAnalysisData.js [hunk 8]
```diff
@@ -46,6 +47,7 @@ const transformedAnalysisData = {
       pmFileName: '',
       analysisFileName: 'analysis-SF1624HT2008_1.pdf',
       roundIdList: '1',
```
Comment (lenaar) [inline test/mocks/transformedAnalysisData.js:49]: `ladokRoundIds ` and `roundIdList` has to be removed everywhere, in order to be sure tests work

**Review by lenaar — CHANGES_REQUESTED** [review]
Need some cleaning of `ladokRoundIds ` and `roundIdList`. It has to be removed everywhere, in order to be sure tests work

File: server/apiCalls/kursPmDataApi.js (hunk 1 above)
Comment (4886Talga) [reply server/apiCalls/kursPmDataApi.js]: reverted back to old code

File: test/mocks/mockCourseAnalysis.js (hunk 2 above)
Comment (4886Talga) [reply test/mocks/mockCourseAnalysis.js]: roundIdList was removed from everywhere

File: test/mocks/mockCourseAnalysis.js (hunk 3 above)
Comment (4886Talga) [reply test/mocks/mockCourseAnalysis.js]: roundIdList was removed from everywhere

File: test/mocks/mockMiniMemos.js (hunk 4 above)
Comment (4886Talga) [reply test/mocks/mockMiniMemos.js]: ladokRoundIds was removed from everywhere

File: test/mocks/mockMiniMemos.js (hunk 6 above)
Comment (4886Talga) [reply test/mocks/mockMiniMemos.js]: ladokRoundIds and roundIdList was removed from everywhere

File: test/mocks/rawAnalysisData.js (hunk 7 above)
Comment (4886Talga) [reply test/mocks/rawAnalysisData.js]: ladokRoundIds and roundIdList was removed from everywhere

File: test/mocks/transformedAnalysisData.js (hunk 8 above)
Comment (4886Talga) [reply test/mocks/transformedAnalysisData.js]: ladokRoundIds and roundIdList was removed from everywhere

**Review by lenaar — APPROVED** [review]
Looks good to me

**Pull request merged**